haproxy/tests/legacy/test_haproxy.py

```diff
@@ -131,7 +131,8 @@ def test_check(aggregator, check, instance):
     _test_service_checks(aggregator, count=0)
 
     aggregator.assert_all_metrics_covered()
-    aggregator.assert_metrics_using_metadata(get_metadata_metrics(), check_submission_type=True)
+    # The assertion below fails due to difference between new and legacy metric types in metadata.
```

Comment:
do we want to keep old code in here? should we find a way to still test via metadata, or get rid of this altogether?